File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

// Including function prototypes
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompting the user for text
    string text = get_string("Text: ");

    // Instantiating functions
    float letters = count_letters(text);
    float words = count_words(text);
    float sentences = count_sentences(text);

    // Calculating index
    float letters_h = (letters / words) * 100;
    float sentences_h = (sentences / words) * 100;

    int index = round(0.0588 * letters_h - 0.296 * sentences_h - 15.8);

    // Printing grade based on index
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

// Counting letters in text
int count_letters(string text)
{
    int letters = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

// Counting words in text
int count_words(string text)
{
    int words = 1;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isspace(text[i]))
        {
            words++;
        }
    }
    return words;
}

// Counting sentances in text
int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

File: sentimental-readability/readability.py
# Readability program
import cs50


def main():

    # Prompting user for text
    text = cs50.get_string("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # Calculating index
    letters_w = (letters / words) * 100
    sentences_w = (sentences / words) * 100

    index = round(0.0588 * letters_w - 0.296 * sentences_w - 15.8)

    # Printing grade based on index
    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")


# Counting letters
def count_letters(text):

    letters = 0
    for i in range(len(text)):
        if text[i].isalpha():
            letters += 1
    return letters


# Counting words
def count_words(text):

    words = 1
    for i in range(len(text)):
        if text[i].isspace():
            words += 1
    return words


# Counting sentances
def count_sentences(text):

    sentences = 0
    for i in range(len(text)):
        if (text[i] == '.') or (text[i] == '!') or (text[i] == '?'):
            sentences += 1
    return sentences


main()
